src/ts/ConvivaAnalyticsTracker.ts

```diff
@@ -112,23 +112,34 @@ export interface EventAttributes {
 export class ConvivaAnalyticsTracker {
   private static readonly VERSION: string = '{{VERSION}}';
 
-  private static readonly STALL_TRACKING_DELAY_MS = 100;
-  private readonly player: PlayerAPI;
-  private readonly events: typeof PlayerEvent;
-  private readonly handlers: PlayerEventWrapper;
+  public static readonly STALL_TRACKING_DELAY_MS = 100;
+  private _player: PlayerAPI;
+
+  private get player(): PlayerAPI {
+    if (!this._player) {
+      throw new Error('Player is not initialized, either pass it to the constructor or attach it via `attachPlayer` before using the integration.');
+    }
+    return this._player;
+  }
+
+  private get isPlayerAttached(): boolean {
+    return !!this._player;
+  }
```

Comment:
The null check is disabled in this repo, the compiler does not complain. Even if there was the null check enabled, you would need to check that the player exists before accessing it.

`isPlayerAttached` is to pronounce the intent better, instead of just checking `if (this._player)`.

The player is not initialized error should never happen in production. Manual + automated testing should prevent that, hopefully.

Reply:
I evaluated if it's easily possible to enable the strict mode, but it will require quite some effort.